File: jdroid-java-remote-config/src/main/java/com/jdroid/java/remoteconfig/RemoteConfigLoader.java
package com.jdroid.java.remoteconfig;

import java.util.List;

public interface RemoteConfigLoader {
	
	void fetch();
	
	Object getObject(RemoteConfigParameter remoteConfigParameter);
	
	String getString(RemoteConfigParameter remoteConfigParameter);

	List<String> getStringList(RemoteConfigParameter remoteConfigParameter);
	
	Boolean getBoolean(RemoteConfigParameter remoteConfigParameter);
	
	Long getLong(RemoteConfigParameter remoteConfigParameter);
	
	Double getDouble(RemoteConfigParameter remoteConfigParameter);
	
}

File: jdroid-java-remote-config/src/main/java/com/dipien/java/remoteconfig/RemoteConfigLoader.kt
package com.dipien.java.remoteconfig

interface RemoteConfigLoader {

    fun fetch()

    fun getObject(remoteConfigParameter: RemoteConfigParameter): Any?

    fun getString(remoteConfigParameter: RemoteConfigParameter): String?

    fun getStringList(remoteConfigParameter: RemoteConfigParameter): List<String>?

    fun getBoolean(remoteConfigParameter: RemoteConfigParameter): Boolean?

    fun getLong(remoteConfigParameter: RemoteConfigParameter): Long?

    fun getDouble(remoteConfigParameter: RemoteConfigParameter): Double?
}
